freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -98,72 +120,256 @@ def _get_requests_session(cls, retry_options={}):
+        and an optional name_stream:name_stream_version mapping.
+
+        :param source_value string: The string in a compose source field.
+        :param name_stream_only boolean: True only generate name_stream set,
+            False also generate a mapping between name_stream and name_stream_version.
+        :return: a tuple of name_stream_set and name_stream:name_stream_version mapping.
+        :rtype: tuple
+        """
+        mapping = {}
+        name_stream_set = set()
+        for source in source_value.split():
+            source = source.strip()
+            name_stream = source.rsplit(":", 2)[0]
+            name_stream_set.add(name_stream)
+            if name_stream_only is False:
+                name_stream_version = source.rsplit(":", 1)[0]
+                mapping[name_stream] = name_stream_version
+        return name_stream_set, mapping
+
+    def _outdated_composes(self, original_odcs_compose_ids, module_name_stream_set):
+        """
+        Generate outdated composes.
+
+        :param original_odcs_compose_ids list: Original compose ids.
+        :param module_name_stream_set set: Module name stream set from an advisory.
+        :return: a set of outdated composes.
+        :rtype: set
+        """
+        outdated_composes = set()
+        for compose_id in original_odcs_compose_ids:
+            compose = self.odcs.get_compose(compose_id)
+            source_type = compose.get("source_type")
+            if source_type != PungiSourceType.MODULE:
+                outdated_composes.add(compose_id)
+                continue
+
+            name_stream_set, _ = self._parse_compose_source(compose.get("source", ""))
+
+            # If all the sources are not in advisory, renew them
+            if name_stream_set.isdisjoint(module_name_stream_set):
+                outdated_composes.add(compose_id)
+
+        return outdated_composes
+
+    def _missing_composes(
+        self,
+        original_odcs_compose_ids,
+        module_name_stream_set,
+        module_name_stream_version_set,
+    ):
+        """
+        Generate missing composes.
+
+        :param original_odcs_compose_ids list: Original compose ids.
+        :param module_name_stream_set set: Module name stream set from an advisory.
+        :param module_name_stream_version_set set: Module name stream version set from an advisory.
+        :return: a set of missing composes.
+        :rtype: set
+        """
+        missing_composes = set()
+        for compose_id in original_odcs_compose_ids:
+            compose = self.odcs.get_compose(compose_id)
+            source_type = compose.get("source_type")
+            if source_type == PungiSourceType.MODULE:
+                name_stream_set, mapping = self._parse_compose_source(compose.get("source", ""), False)
+
+                # If the two sets are joint, union compose's name_stream_version set
+                # and advisory module's name_stream_version set in missing composes.
```

Comment:
No need for comment - it explains what the code does instead of why - the code is clear enough.

```suggestion
```

Reply:
Ack. Removed